pkg/controllers/provisioning/v1alpha1/reallocation/terminate.go

```diff
@@ -30,39 +30,50 @@ type Terminator struct {
 	cloudprovider cloudprovider.Factory
 }
 
-func (t *Terminator) CordonNodes(ctx context.Context, nodes []*v1.Node) error {
+// AddTTLs takes in a list of underutilized nodes and adds TTL to them
+func (t *Terminator) AddTTLs(ctx context.Context, nodes []*v1.Node) error {
+	if len(nodes) == 0 {
```

Comment:
If you ttl within the for loop, you don't need to exit here.

Reply:
We can get rid of the check here, since we log TTL added per node.